discovery-provider/src/utils/ipfs_lib.py

```diff
@@ -107,11 +108,27 @@ def query_ipfs_metadata_json(self, gateway_ipfs_urls, metadata_format):
                     logger.error(f"IPFSClient | {url} generated an exception: {exc}")
         return formatted_json
 
-    def get_metadata_from_gateway(self, multihash, metadata_format):
+    def get_metadata_from_gateway(self, multihash, metadata_format, user_replica_set=None):
```

Comment:
maybe worth documenting that user_replica_set here is the "," separated string version

Reply:
added